Heimdall/Heimdall.swift

```diff
@@ -882,18 +894,21 @@ private extension NSData {
                 offset += OID.count
                 
                 // Type
-                if bytes[offset++] != 0x03 {
+                if bytes[offset] != 0x03 {
```

Comment:
This should be correct. If this is true, then the `if` statement does `return self` and there is no point in incrementing the `offset`, as the entire function exits.